tests/integ/modin/index/test_equals.py

```diff
@@ -10,28 +10,77 @@
 from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
 
 
+def to_snow_index(native_index: native_pd.Index):
+    """
+    Convert native pandas index to snowpark pandas index.
+    Args:
+        native_index: Native pandas index.
+
+    Returns:
+        Snowpark pandas index.
+    """
+    if isinstance(native_index, native_pd.DatetimeIndex):
+        return pd.DatetimeIndex(native_index)
+    return pd.Index(native_index)
+
+
 @pytest.mark.parametrize(
     "lhs, rhs, expected",
     [
-        ([], [], True),  # empty indices
-        ([None], [None], True),  # none indices
-        ([1, 2, 3], [1, 2, 3], True),
-        ([1, 2, None], [1, 2, None], True),  # nulls are considered equal
-        ([1, 2, 3], [1.0, 2.0, 3.0], True),  # type is ignored
-        ([1, 2, 3], [1, 3, 2], False),  # different order
-        ([1, 2, 3], [1, 2, 3, 4], False),  # extra value in right
-        ([1, 2, 3, 4], [1, 2, 3], False),  # extra value in left
+        (native_pd.Index([]), native_pd.Index([]), True),  # empty indices
+        (native_pd.Index([None]), native_pd.Index([None]), True),  # none indices
+        (native_pd.Index([1, 2, 3]), native_pd.Index([1, 2, 3]), True),
+        (
+            native_pd.Index([1, 2, None]),
+            native_pd.Index([1, 2, None]),
+            True,
+        ),  # nulls are equal
+        (
+            native_pd.Index([1, 2, 3]),
+            native_pd.Index([1.0, 2.0, 3.0]),
+            True,
+        ),  # type is ignored
+        (
+            native_pd.Index([1, 2, 3]),
+            native_pd.Index([1, 3, 2]),
+            False,
+        ),  # different order
+        (
+            native_pd.Index([1, 2, 3]),
+            native_pd.Index([1, 2, 3, 4]),
+            False,
+        ),  # extra value in right
+        (
+            native_pd.Index([1, 2, 3, 4]),
+            native_pd.Index([1, 2, 3]),
+            False,
+        ),  # extra value in left
+        (
+            native_pd.DatetimeIndex(["2024-01-01 03:00:00+00:00"]),
+            native_pd.DatetimeIndex(["2024-01-01 03:00:00+00:00"]),
+            True,
+        ),  # same
+        (
+            native_pd.DatetimeIndex(["2024-01-01 04:00:00+00:00"]),
+            native_pd.DatetimeIndex(["2024-01-01 05:00:00+00:00"]),
+            False,
+        ),  # different
+        (
+            native_pd.DatetimeIndex(["2024-01-01 04:00:00+00:00"]),
+            native_pd.DatetimeIndex(["2024-01-01 04:00:00+05:00"]),
+            False,
+        ),  # different tz
     ],
 )
 def test_index_equals(lhs, rhs, expected):
     with SqlCounter(query_count=2, join_count=2):
         native_result = native_pd.Index(lhs).equals(native_pd.Index(rhs))
         assert native_result == expected
 
-        snow_result = pd.Index(lhs).equals(pd.Index(rhs))
+        snow_result = to_snow_index(lhs).equals(pd.Index(rhs))
```

Comment:
we don't have to change this right? `pd.Index(pd.DatetimeIndex(...))` should be a `DatetimeIndex`

Reply:
done